projects/core/components/primitive-calendar/pipes/convert-to-calendar-sheet/utils.ts

```diff
@@ -0,0 +1,78 @@
+import {
+    DAYS_IN_WEEK,
+    inRange,
+    tuiAssert,
+    TuiDay,
+    TuiDayOfWeek,
+    TuiMonth,
+} from '@taiga-ui/cdk';
+
+/**
+ * Computes day of week offset of the beginning of the month
+ */
+const getMonthStartDaysOffset = (
+    month: TuiMonth,
+    firstDayOfWeek: TuiDayOfWeek,
+): number => {
+    const startMonthOffsetFromSunday = new Date(month.year, month.month, 1).getDay();
+
+    return startMonthOffsetFromSunday >= firstDayOfWeek
+        ? startMonthOffsetFromSunday - firstDayOfWeek
+        : DAYS_IN_WEEK - (firstDayOfWeek - startMonthOffsetFromSunday);
+};
+
+/*
+TODO delete:
```

Comment:
let's point out that it is for 3.0 version to find it easily later. For example `TODO: delete in v3.0`

Reply:
added it